zcash_client_backend/src/data_api.rs

```
@@ -2597,6 +2607,21 @@ pub trait WalletWrite: WalletRead {
         key_source: Option<&str>,
     ) -> Result<Self::Account, Self::Error>;
 
+    /// Imports the given pubkey into the account without key derivation information, and adds the
+    /// associated transparent p2pkh address.
+    ///
+    /// The imported address will contribute to the balance of the account, but spending funds held
+    /// by this address requires the associated spending keys to be provided explicitly when
+    /// calling [`create_proposed_transactions`].
+    ///
+    /// [`create_proposed_transactions`]: wallet::create_proposed_transactions
+    #[cfg(feature = "transparent-inputs")]
+    fn import_standalone_transparent_pubkey(
```

In a pairing with @nuttycom, we (believe that we) decided to add a new "constructor" API that allows for specifically creating legacy transparent-only accounts, and then this API should return an error if `account` is not one of those (to prevent "bleeding" of transparent addresses into the account semantics of existing accounts).

We also discussed how P2SH multisig will work, and that will be another separate constructor for that which takes in the descriptor template (alongside the relevant viewing-level extpubkey for this wallet's spending authority), and creates an account that is transparent-only but otherwise has a consistent and diversifiable address pool (since the proposed ZIP 48 mechanism supports deriving a sequence of P2SH addresses).

---

The notion that we should add a separate constructor API for legacy transparent-only accounts is the most disputed point here; we've carried on the discussion and need to consider the implications if someone imports the keys from two separate `wallet.dat` files that both hold funds in legacy addresses. In this case, since we don't want to brick those funds, my proposal is that we should make the zcashd ZIP 32 legacy account index `0x7FFFFFFF` the sole determiner of whether legacy semantics are applied to an account. Then, for any RPC method that we choose to support that involves legacy semantics, we add an optional `account_uuid` parameter. If only a single account having ZIP 32 index `0x7FFFFFF` exists in the wallet, then legacy semantics apply to that account; if more than one such account exists, then the user MUST specify the account UUID to disambiguate.

Given this approach, a separate constructor for creating the legacy account is not necessary; instead, we would just perform a check against the ZIP 32 account identifier of the account being specified in this API call and return an error if it is not 0x7FFFFFF.

This _does_ mean that we would not support import of transparent spending keys into the wallet _unless_ a legacy account exists; I think that's an okay restriction for right now.